misaligned.cpp

```diff
@@ -1,21 +1,63 @@
 #include <iostream>
 #include <assert.h>
+#include <map>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+
+int calculatePairNo(int majorIndex, int minorIndex)
+{
+    return ((majorIndex * 5 + minorIndex) + 1); 
+}
+
+std::string createColorPair(int majorIndex, int minorIndex)
+{
+    std::string majorColorName(majorColor[majorIndex]);
+    std::string minorColorName(minorColor[minorIndex]);
+    return majorColorName.append(" | ").append(minorColorName);
+}
+
+int getPairNumber(int majorIndex, int minorIndex, std::map<int, std::string>& ColorPairs)
+{
+    int pair_no = calculatePairNo(majorIndex, minorIndex);

```

Comment:
here the pair number is one based. Maybe renaming the variable from pair_no to one_based_pair_no could be helpful.

Reply:
Could I know what is one based?

Reply:
Indexing an array is zero based which means index starts from zero. In some structures index starts from 1, then it is one based. While displaying pair number to user, pair number zero has little meaning. Numbering from 1 makes sense to users. That is what I meant by 1 based. Hope it's clear.